File: 0190-ReverseBits/Solution.java
import java.util.*;

public class Solution {
    
    public int reverseBits(int n) {
        int reversed = n & 1;
        for (int i = 0; i < 31; i++) {
            n >>>= 1; // unsigned right shift
            reversed <<= 1;
            reversed |= n & 1;
        }
        return reversed;
    }

    public static void main(String[] args) {
        System.out.println();
    }

}

File: 0190-ReverseBits/solution.py
class Solution:

    def reverseBits(self, n: int) -> int:
        res = 0
        for i in range(31):
            b = n & 1
            res |= b
            n >>= 1
            res <<= 1
        res |= n
        return res


if __name__ == "__main__":
    s = Solution()

